physics/MP/Thompson/module_mp_thompson.F90

```diff
@@ -1025,223 +1023,223 @@ SUBROUTINE mp_gt_driver(qv, qc, qr, qi, qs, qg, ni, nr, nc,     &
-      INTEGER:: kmax_qc,kmax_qr,kmax_qi,kmax_qs,kmax_qg,kmax_ni,kmax_nr
-      INTEGER:: i_start, j_start, i_end, j_end
-      LOGICAL, OPTIONAL, INTENT(IN) :: diagflag
-      INTEGER, OPTIONAL, INTENT(IN) :: do_radar_ref
-      logical :: melti = .false.
-      INTEGER :: ndt, it
-
-      ! CCPP error handling
-      character(len=*), optional, intent(  out) :: errmsg
-      integer,          optional, intent(  out) :: errflg
-
-      ! CCPP
-      if (present(errmsg)) errmsg = ''
-      if (present(errflg)) errflg = 0
-
-      ! No need to test for every subcycling step
-      test_only_once: if (first_time_step .and. istep==1) then
-         ! Activate this code when removing the guard above
-   
-         if ( (present(tt) .and. (present(th) .or. present(pii))) .or. &
-              (.not.present(tt) .and. .not.(present(th) .and. present(pii))) ) then
-            if (present(errmsg) .and. present(errflg)) then
-               write(errmsg, '(a)') 'Logic error in mp_gt_driver: provide either tt or th+pii'
-               errflg = 1
-               return
-            else
-               write(*,'(a)') 'Logic error in mp_gt_driver: provide either tt or th+pii'
-               stop
+         real(wp), dimension(its:ite, jts:jte):: pcp_ra, pcp_sn, pcp_gr, pcp_ic
+         real(wp) :: dt, pptrain, pptsnow, pptgraul, pptice
+         real(wp) :: qc_max, qr_max, qs_max, qi_max, qg_max, ni_max, nr_max
+         integer:: lsml
+         real(wp) :: rand1, rand2, rand3, rand_pert_max
+         integer:: i, j, k, m
+         integer:: imax_qc,imax_qr,imax_qi,imax_qs,imax_qg,imax_ni,imax_nr
+         integer:: jmax_qc,jmax_qr,jmax_qi,jmax_qs,jmax_qg,jmax_ni,jmax_nr
+         integer:: kmax_qc,kmax_qr,kmax_qi,kmax_qs,kmax_qg,kmax_ni,kmax_nr
+         integer:: i_start, j_start, i_end, j_end
+         logical, optional, intent(in) :: diagflag
+         integer, optional, intent(in) :: do_radar_ref
+         logical :: melti = .false.
+         integer :: ndt, it
+
+         ! CCPP error handling
+         character(len=*), optional, intent(  out) :: errmsg
+         integer,          optional, intent(  out) :: errflg
+
+         ! CCPP
+         if (present(errmsg)) errmsg = ''
+         if (present(errflg)) errflg = 0
+
+         ! No need to test for every subcycling step
+         test_only_once: if (first_time_step .and. istep==1) then
+            ! Activate this code when removing the guard above
+      
+            if ( (present(tt) .and. (present(th) .or. present(pii))) .or. &
+               (.not.present(tt) .and. .not.(present(th) .and. present(pii))) ) then
+               if (present(errmsg) .and. present(errflg)) then
+                  write(errmsg, '(a)') 'Logic error in mp_gt_driver: provide either tt or th+pii'
+                  errflg = 1
+                  return
+               else
+                  write(*,'(a)') 'Logic error in mp_gt_driver: provide either tt or th+pii'
+                  stop
```

Comment:
Here is a stop statement to clean up in the future. CCPP schemes shouldn't use STOP and should do what is done above with errmsg, errflg, return.

Reply:
OK, nevermind, I see that this should never execute in the CCPP.